src/app/core/interceptors/http-error.interceptor.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor,
+  HttpErrorResponse
+} from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { retry, catchError } from 'rxjs/operators';
+import { AuthService } from '../services/auth.service';
+
+@Injectable()
+export class HttpErrorInterceptor implements HttpInterceptor {
+
+  constructor(private authService: AuthService) { }
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
```

Comment:
Dlaczego interceptor a nie error handler? 
Interceptor też jest ok, ale np. logika związana z logowaniem / rejestracją - no tutaj nie pasuje. bo to nie jest coś co występuje w wielu rzeczach. i taka obsługa błędóe to powinna być w serwisie logowania/rejestracji.

Reply:
Tutaj powinny znaleźć się rzeczy uniwersalne. Globalny error handling błędów js, błędów api, itp. czyli rzeczy, które albo pomijasz albo wystarczy je ogarnąć generycznie.

Reply:
done